database/migrations/2023_01_20_184154_create_notification_virtual_column_url.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateNotificationVirtualColumnUrl extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('notifications', function (Blueprint $table) {
+            $table->string('url')->virtualAs('JSON_UNQUOTE(data->"$.url")')->index();
```

Comment:
Virtual columns are calculated in every read. It will be better to use a "real" mysql column for this info. The attribute "url" of a notification is never changed after it is created